batchflow/models/tf/mobilenet.py

```diff
@@ -189,38 +232,112 @@ def body(cls, inputs, name='body', **kwargs):
-        x = conv_block(inputs, 'cna Cna cn', conv_filters, [1, 3, 1], name=name, strides=[1, strides, 1], **kwargs)
+        x = conv_block(inputs, 'cnawna', [num_filters, num_filters], [1, kernel_size], strides=[1, strides],
+                       name='%s-exp' % name, **kwargs)
+
+        if se_block:
+            if not isinstance(se_block, dict):
+                se_block = dict(activation=[kwargs.get('activation', tf.nn.relu), h_sigmoid], ratio=num_filters // 4)
+            x = cls.se_block(x, **se_block, name='%s-se' % name, data_format=kwargs.get('data_format'))
+
+        x = conv_block(x, 'cn', filters, 1, name='%s-down' % name, **kwargs)
+
         if residual:
             x = inputs + x
         return x
+
+
+class MobileNet_v3(MobileNet):
+    """ MobileNet version 3 architecture """
+    @classmethod
+    def default_config(cls):
+        config = TFModel.default_config()
+        config['initial_block'].update(dict(layout='cna', filters=16, kernel_size=3, strides=2, activation=h_swish))
+        config['body'].update(dict(width_factor=1, layout=deepcopy(_V3_LARGE_DEFAULT_BODY)))
+        config['head'].update(dict(layout='cnavcacV', filters=[960, 1280, 2], pool_size=7, kernel_size=1,
+                                   activation=h_swish))
+        config['loss'] = 'ce'
+
+        return config
+
+
+class MobileNet_v3_small(MobileNet):
+    """ MobileNet version 3 small architecture """
+    @classmethod
+    def default_config(cls):
+        config = TFModel.default_config()
+        config['initial_block'].update(dict(layout='cna', filters=16, kernel_size=3, strides=2, activation=h_swish))
+        config['body'].update(dict(width_factor=1, layout=deepcopy(_V3_SMALL_DEFAULT_BODY)))
+        config['head'].update(dict(layout='cnavcacV', filters=[576, 1280, 2], pool_size=7,
+                                   kernel_size=1, activation=h_swish,
+                                   se_block=dict(activation=[h_swish, h_sigmoid], ratio=144)))
+        config['loss'] = 'ce'
+
+        return config
+
+    @classmethod
+    def head(cls, inputs, se_block=None, name='head', **kwargs):
+        """ The last network layers which produce predictions
+
+        Parameters
+        ----------
+        inputs : tf.Tensor
+            input tensor
+        name : str
+            scope name
+
+        Returns
+        -------
+        tf.Tensor
+        """
+        kwargs = cls.fill_params('head', **kwargs)
+        layout = kwargs.pop('layout')
+        filters = kwargs.pop('filters')
+        if kwargs.get('se_block'):
+            x = conv_block(inputs, layout=layout[:3], filters=filters[0], name='%s-conv1' % name, **kwargs)
+            x = cls.se_block(x, **se_block, data_format=kwargs.get('data_format'), name='%s-se' % name)
+            x = conv_block(x, layout=layout[3:], filters=filters[1:], name='%s-conv2' % name, **kwargs)
+            return x
+        return conv_block(inputs, layout=layout, filters=filters, name=name, **kwargs)
```

Comment:
To remove complicated implicit configuration, hardcode all the parameters (layout, filters) from the article right in the `head` method.
Later we will add the possibility to chain methods.

Reply:
hardcoded layout, left filters as parameter modified in build_config.